### PR TITLE
Add class distribution for segmentation demo results

### PR DESCRIPTION
**Pull request by eizamaliev**
With `-r` option on, both C++ and Python segmentation demo print raw result as class distribution, e.g.:
```
[ INFO ]  Class ID | Pixels | Percentage
[ INFO ]        #0 | 870304 | 94.434%
[ INFO ]        #1 |  51296 | 5.56597%
```

TODO:
- [x] Add labels for cpp/segmentation model
- [x] Update documentation

### CODE REVIEW

File: demos/common/cpp/models/src/segmentation_model.cpp [hunk 1]
```diff
@@ -22,6 +22,26 @@ using namespace InferenceEngine;
...
+        while (std::getline(inputFile, label)) {
+            labelsList.push_back(label);
+        }
+        if (labelsList.empty())
+            throw std::logic_error("File is empty: " + labelFilename);
+    }
+
+    return labelsList;
+}
+
```
Comment (fzhar) [inline demos/common/cpp/models/src/segmentation_model.cpp:44]: This is 3rd separate implementation of loadLabels (we already have 2 in DetectionModel and Classification model). We definetely have to unify them as single function in ImageModel class. May not in this patch but in separate one.

File: demos/segmentation_demo/cpp/main.cpp [hunk 2]
```diff
@@ -188,6 +194,33 @@ cv::Mat renderSegmentationData(const ImageResult& result, OutputTransform& outpu
     return output;
 }
 
+void print_raw_results(const ImageResult& result, std::vector<std::string> labels)
```
Comment (fzhar) [inline demos/segmentation_demo/cpp/main.cpp:197]: I'd suggest to print image output resolution and total number of pixels here (before statistics). That would ease understanding of what statistics base is used - is is network output or input image resolution.

**Review by fzhar — CHANGES_REQUESTED** [review]

File: demos/segmentation_demo/cpp/main.cpp [hunk 3]
```diff
@@ -188,6 +194,33 @@ cv::Mat renderSegmentationData(const ImageResult& result, OutputTransform& outpu
...
+    for (int i = 0; i < max_classes; ++i)
+    {
+        const int value = histogram.at<int>(i);
+        if (value > 0)
+        {
+            std::string label = i < labels.size() ? labels[i] : "#" + std::to_string(i);
+            slog::info << " "
+                << std::setw(7) << label << " | "
+                << std::setw(6) << histogram.at<float>(i) << " | "
+                << std::setw(6) << histogram.at<float>(i) / all * 100 << "%"
```
Comment (Wovchena) [inline demos/segmentation_demo/cpp/main.cpp:218]: You should fix precision similar to python

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: demos/common/cpp/models/src/segmentation_model.cpp (hunk 1 above)
Comment (eizamaliev) [reply demos/common/cpp/models/src/segmentation_model.cpp:44]: I totally agree with, but not now.

File: demos/segmentation_demo/cpp/main.cpp (hunk 2 above)
Comment (eizamaliev) [reply demos/segmentation_demo/cpp/main.cpp:197]: I would agree with you, but similar function in `object_detection_demo` doesnt print such info, assuming that it depends on image, not network, resolution.

File: demos/segmentation_demo/cpp/main.cpp [hunk 4]
```diff
@@ -188,6 +194,33 @@ cv::Mat renderSegmentationData(const ImageResult& result, OutputTransform& outpu
...
+    cv::calcHist(&result.resultImage, 1, 0, cv::Mat(), histogram, 1, &max_classes, &ranges);
+    const double all = result.resultImage.cols * result.resultImage.rows;
+    for (int i = 0; i < max_classes; ++i)
+    {
+        const int value = histogram.at<int>(i);
+        if (value > 0)
+        {
+            std::string label = i < labels.size() ? labels[i] : "#" + std::to_string(i);
+            slog::info << " "
+                << std::setw(7) << label << " | "
```
[the remark this answers is not in the document]
Comment (eizamaliev) [reply demos/segmentation_demo/cpp/main.cpp]: done

File: demos/segmentation_demo/cpp/main.cpp [hunk 5]
```diff
@@ -188,6 +194,33 @@ cv::Mat renderSegmentationData(const ImageResult& result, OutputTransform& outpu
...
+void print_raw_results(const ImageResult& result, std::vector<std::string> labels)
+{
+
+    slog::info << " Class ID | Pixels | Percentage " << slog::endl;
+    cv::Mat histogram;
+    double min_val, max_val;
+    cv::minMaxLoc(result.resultImage, &min_val, &max_val);
+    int max_classes = static_cast<int>(max_val) + 1; // We use +1 for only background case
+    const float range[] = { 0, max_classes };
+    const float * ranges = { range };
```
[the remark this answers is not in the document]
Comment (eizamaliev) [reply demos/segmentation_demo/cpp/main.cpp:205]: Are you sure about this?

File: demos/segmentation_demo/cpp/main.cpp (hunk 3 above)
Comment (eizamaliev) [reply demos/segmentation_demo/cpp/main.cpp]: Done

File: demos/segmentation_demo/cpp/main.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (Wovchena) [reply demos/segmentation_demo/cpp/main.cpp:205]: The suggestion is from OpenCV example

File: demos/segmentation_demo/cpp/main.cpp [hunk 6]
```diff
@@ -188,6 +194,34 @@ cv::Mat renderSegmentationData(const ImageResult& result, OutputTransform& outpu
...
+{
+    slog::info << "     Class ID     | Pixels | Percentage " << slog::endl;
+
+    double min_val, max_val;
+    cv::minMaxLoc(result.resultImage, &min_val, &max_val);
+    int max_classes = static_cast<int>(max_val) + 1; // We use +1 for only background case
+    const float range[] = { 0, static_cast<float>(max_classes) };
+    const float * ranges = { range };
+    cv::Mat histogram;
+    cv::calcHist(&result.resultImage, 1, 0, cv::Mat(), histogram, 1, &max_classes, &ranges);
```
Comment (Wovchena) [inline demos/segmentation_demo/cpp/main.cpp:207]:
```suggestion
    cv::calcHist(&result.resultImage, 1, 0, cv::Mat(), histogram, 1, &max_classes, ranges);
```

File: demos/segmentation_demo/cpp/main.cpp (hunk 3 above)
Comment (Wovchena) [reply demos/segmentation_demo/cpp/main.cpp]: They are still different. It should be
`<< std::setw(5) << std::setprecision(2) << std::fixed << std::right << value / all * 100 << "%"`

Comment (eizamaliev) [reply demos/segmentation_demo/cpp/main.cpp]: Thanks, done!

**Review by Wovchena — APPROVED** [review]

**Comment by vladimir-dudnik** [comment]
12:31:37  /jenkins/workspace/omz/precommit/demos_ubuntu18/src/demos/segmentation_demo/cpp/main.cpp:215:35: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]

**Comment by vladimir-dudnik** [comment]
@eizamaliev bump?

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by Wovchena** [comment]
Jenkins please retry a build

File: demos/segmentation_demo/cpp/main.cpp (hunk 2 above)
Comment (fzhar) [reply demos/segmentation_demo/cpp/main.cpp:197]: Here you print number of pixels - but it is not clear - 10 pixels - is it many or not ? We need to know resolution to evaluate that. I wouldn't insist on this change as percentage is printed, but I'd suggest to put such info here (and maybe add it to object_detection_demo too)

**Review by fzhar — APPROVED** [review]

**Pull request merged**